### DISCUSSION TITLE
week # problem

### DISCUSSION THREAD

**Discussion by stroebie** [kimai/kimai#2872] [Questions & Answers]
I am a new in Kimai, and I am investigation a problem reported by our colleagues

We have 2 Kimai installations, our current production environment, running version 1.6.2 and a new test environment running 1.14.3.

If I select the calendar view in our production environment (1.6) then I see that the week starts on a Sunday and surprisingly I see that the current week is set on week 43. (on my Samsung today is week 42)

If I select the calendar view in our test environment (1.14) then I see that the week starts on a Monday with the current week set on 42… which is correct.

My questions: what settings do I have to turn to make the calendar view in our production environment (1.6) start on a Monday (iso a Sunday) ? and secondly which parameter do I have to turn the make the week number correct and set to 42 i.s.o. 43?
 
Thanks a lot in advance for your support and help.

**Comment by kevinpapst** [comment]
Unfortunately this is one of the features that does not work out-of-the-box, but is rather complicated to build.
Try the demo with 1.15.5, where I fixed yet another "Sunday is the first day of the week" problem.

You can choose the first day of the week in user preferences, it is not a global setting becuase each user could live somewhere else.

And your prod environment 1.6 will definitely need an upgrade, I donÄt think this feature was available back then.

**Reply by stroebie** [reply]
Kevin, thank you for your reply. 
We know that we running a too old Kimai kernel. We plan to upgrade to a newer Kimai version. My predecessor has already build an environment around the Kimai 1.14 kernel. 

An answer that is not clear to me is about the week numbers. The week number in 1.6 is not correct in the calendar view. Is that a parameter that we have to set somewhere or is that a flaw in the 1.6 kernel that has been taken care of in later Kimai versions? In 1.14 the week number is correct. 

Is there somewhere a kind of a release note where the differences ( bug fixed, new feature, improvements) between the versions are listed? I could not find anything in the documentation.

**Reply by kevinpapst** [reply]
Read through the release notes: https://github.com/kevinpapst/kimai2/releases

This is 1.6 https://github.com/kevinpapst/kimai2/releases/tag/1.6
It was released 2 years ago, so don't expect me to know what happened since then ... way too much.

You have to upgrade, there is no other way / quick fix. The longer you wait, the worse it gets.
